### PR TITLE
Add WfP namespace resource

### LINKED ISSUE #2426: Workers for Platforms
### Current Terraform and Cloudflare provider version

Terraform v1.4.6
Cloudflare v4.5.0

### Description

Ability to create WFP dispatch namespaces, and bind dispatch namespaces to a worker

### Use cases

To use WFP with workers

### Potential Terraform configuration

```hcl
resource "cloudflare_workers_for_platform_namespace" "my_wfp_namespace" {
  name = "my-dispatch-namespace" 
}

resource "cloudflare_worker_script" "my_script" {
  dispatch_namespaces_binding {
    name      = "MY_BINDING"
    namespace = "MY_NAMESPACE"
  }
}
```

### References

_No response_

### PR DESCRIPTION
**Pull request by WalshyDev**
Implements a new Workers for Platforms namespace resource which allows users to manage their WfP namespace in Terraform. With this change, you can _almost_ have everything within Terraform! We're just missing the [Dispatch Worker](https://developers.cloudflare.com/cloudflare-for-platforms/workers-for-platforms/get-started/dynamic-dispatch/) due to the namespace binding not being fully implemented in cloudflare-go (next up on my list).
Closes #2426

random note: I like this new framework, feels much nicer than sdkv2

Acceptance tests:
```
=== RUN   TestAccCloudflareWorkersForPlatforms_NamespaceManagement
=== PAUSE TestAccCloudflareWorkersForPlatforms_NamespaceManagement
=== RUN   TestAccCloudflareWorkersForPlatforms_UploadUserWorker
=== PAUSE TestAccCloudflareWorkersForPlatforms_UploadUserWorker
=== CONT  TestAccCloudflareWorkersForPlatforms_NamespaceManagement
=== CONT  TestAccCloudflareWorkersForPlatforms_UploadUserWorker
--- PASS: TestAccCloudflareWorkersForPlatforms_NamespaceManagement (3.57s)
--- PASS: TestAccCloudflareWorkersForPlatforms_UploadUserWorker (7.66s)
PASS
ok  	github.com/cloudflare/terraform-provider-cloudflare/internal/framework/service/workers_for_platforms	8.306s
```

Note for acceptance tests: You will need Workers for Platforms enabled, `workers_for_saas` flipper flag.

### CODE REVIEW

**Comment by jacobbednarz** [comment]
acceptance tests all passing

```
TF_ACC=1 go test ./internal/framework/service/workers_for_platforms -v -run "^TestAccCloudflareWorkersForPlatforms_" -count 1 -timeout 120m -parallel 1
=== RUN   TestAccCloudflareWorkersForPlatforms_NamespaceManagement
=== PAUSE TestAccCloudflareWorkersForPlatforms_NamespaceManagement
=== RUN   TestAccCloudflareWorkersForPlatforms_UploadUserWorker
=== PAUSE TestAccCloudflareWorkersForPlatforms_UploadUserWorker
=== CONT  TestAccCloudflareWorkersForPlatforms_NamespaceManagement
--- PASS: TestAccCloudflareWorkersForPlatforms_NamespaceManagement (9.41s)
=== CONT  TestAccCloudflareWorkersForPlatforms_UploadUserWorker
--- PASS: TestAccCloudflareWorkersForPlatforms_UploadUserWorker (11.76s)
PASS
ok  	github.com/cloudflare/terraform-provider-cloudflare/internal/framework/service/workers_for_platforms	22.025s
```

File: internal/framework/service/workers_for_platforms/resource.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+func (r *WorkersForPlatformsResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	idparts := strings.Split(req.ID, "/")
+	if len(idparts) != 2 {
+		resp.Diagnostics.AddError("error importing Workers for Platforms namespace", "invalid ID specified. Please specify the ID as \"account_id/name\"")
+		return
+	}
+	resp.Diagnostics.Append(resp.State.SetAttribute(
+		ctx, path.Root("account_id"), idparts[0],
+	)...)
+	// TODO: Figure out if I can set name here, ID is a UUID and isn't one we really display/use user-facing wise
```
Comment (jacobbednarz) [inline internal/framework/service/workers_for_platforms/resource.go:130]: is this still a TODO? you can set anything you want as the ID however, it shouldn't be based on anything the user can change (i.e. the name). if you must use a use configurable, you'll need to make sure we force replace it whenever it changes like you already have defined in the schema.

Comment (WalshyDev) [reply internal/framework/service/workers_for_platforms/resource.go:130]: Oops nah forgot to remove it. It's using the name now.

Comment (WalshyDev) [inline internal/framework/service/workers_for_platforms/resource.go:130]:
```suggestion
```

**Comment by jacobbednarz** [comment]
thanks! glad to hear you're preferring the plugin framwork 🙇

**Pull request merged**